src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/UnixExportProvider.cs

```diff
@@ -84,482 +82,81 @@ public void Dispose()
-                certBags.AsSpan(0, certCount).Clear();
-                keyBags.AsSpan(0, certCount).Clear();
-                ArrayPool<AttributeAsn>.Shared.Return(certAttrs);
-                ArrayPool<CertBagAsn>.Shared.Return(certBags);
-                ArrayPool<SafeBagAsn>.Shared.Return(keyBags);
-
-                if (encodedAuthSafe.Array != null)
-                {
-                    CryptoPool.Return(encodedAuthSafe);
-                }
-            }
-        }
-
-        private void BuildBags(
-            ICertificatePalCore certPal,
-            ReadOnlySpan<char> passwordSpan,
-            AsnWriter tmpWriter,
-            CertBagAsn[] certBags,
-            AttributeAsn[] certAttrs,
-            SafeBagAsn[] keyBags,
-            ref int certIdx,
-            ref int keyIdx)
-        {
-            tmpWriter.WriteOctetString(certPal.RawData);
-
-            certBags[certIdx] = new CertBagAsn
-            {
-                CertId = Oids.Pkcs12X509CertBagType,
-                CertValue = tmpWriter.Encode(),
-            };
-
-            tmpWriter.Reset();
-
-            if (certPal.HasPrivateKey)
-            {
-                byte[] attrBytes = new byte[6];
-                attrBytes[0] = (byte)UniversalTagNumber.OctetString;
-                attrBytes[1] = sizeof(int);
-                MemoryMarshal.Write(attrBytes.AsSpan(2), in keyIdx);
-
-                AttributeAsn attribute = new AttributeAsn
-                {
-                    AttrType = Oids.LocalKeyId,
-                    AttrValues = new ReadOnlyMemory<byte>[]
-                    {
-                        attrBytes,
-                    }
-                };
-                keyBags[keyIdx] = new SafeBagAsn
-                {
-                    BagId = Oids.Pkcs12ShroudedKeyBag,
-                    BagValue = ExportPkcs8(certPal, passwordSpan),
-                    BagAttributes = new[] { attribute }
-                };
-
-                // Reuse the attribute between the cert and the key.
-                certAttrs[certIdx] = attribute;
-                keyIdx++;
             }
 
-            certIdx++;
-        }
-
-        private static unsafe ArraySegment<byte> EncodeAuthSafe(
-            AsnWriter tmpWriter,
-            SafeBagAsn[] keyBags,
-            int keyCount,
-            CertBagAsn[] certBags,
-            AttributeAsn[] certAttrs,
-            int certIdx,
-            ReadOnlySpan<char> passwordSpan)
-        {
-            string? encryptionAlgorithmOid = null;
-            bool certsIsPkcs12Encryption = false;
-            string? certsHmacOid = null;
-
-            ArraySegment<byte> encodedKeyContents = default;
-            ArraySegment<byte> encodedCertContents = default;
-
-            try
-            {
-                if (keyCount > 0)
-                {
-                    encodedKeyContents = EncodeKeys(tmpWriter, keyBags, keyCount);
-                }
-
-                Span<byte> salt = stackalloc byte[16];
```

Comment:
This is one of the "observable" differences from this implementation. Before this change, the salt was 16 bytes. `Pkcs12Builder` uses the size of the MAC function. In this case, it would be 20 bytes. So the salt size increased by 4 bytes.